sunpy/net/vso/vso.py

```diff
@@ -77,6 +77,13 @@ def iter_records(response):
         yield from prov_item.record.recorditem
 
 
+def iter_sort_response(response):
+    for prov_item in response.provideritem:
+        if not hasattr(prov_item, 'record') or not prov_item.record:
+            continue
+        yield from sorted(prov_item.record.recorditem, key=lambda x: x.time.start)
```

Comment:
If I am reading this right this is assuming that if a `provideritem` block has a `record` subblock, then that has a `recorditem` and that has a `time` with a `start`.

I might just run this past the VSO team to make sure this is a safe assumption.

Reply:
I just checked it for 3-4 types of VSO queries, and it didn't failed so assumed. It would be great if it can be confirmed with VSO if it is the case always.

Reply:
@Cadair I can't find the reference in their docs.

Reply:
I just checked that we do assume in `merge` that `record` will have a `recorditem`, and in their docs it is written that record item will have `time` with `start` in it. Just not sure that will `provideritem` will have `record` in it. Should I assume it will have? Or just sort by any algorithm and use continue with `hasattr(record)` ?

Reply:
`provideritem` might not have a `record` subblock, or it might have an empty block.

The main case when this would happen is if there's an error message from the provider, and use that mechanism to return a warning message from the JSOC when it switches into 'summary row' mode because you asked for something that matched too many records.

You should be able to trigger it by asking for 3 hours worth of AIA data.  Here's an example from IDL:

    IDL> a = vso_search( '2011-01-01','2011-01-01 03:00', inst='aia' );
    Records Returned : JSOC : 0/
        JSOC : Large Query (over 5000 records); Summary rows returned; Each record describes multiple observations
    Records Returned : JSOC : 7202/7202
    IDL> help, a
    A               STRUCT    = -> VSORECORD Array[174]

There were talks in years past about including model results in the VSO, and so it's possible that in the future, not all data will have a time associated with it.  (but I lean towards having it like 'quicklook' data where you have to specifically request it if we don't put it as a completely different method to keep it isolated from observational data)

Reply:
I should mention -- that return structure should've had two `provideritem`s, one with no records, the other with 174.

Reply:
Thanks for the input @jhourcle 

> provideritem might not have a record subblock, or it might have an empty block.

Cool, I think we handle this.

Maybe it is worth just guarding this whole block in an `except AttributeError` so that if there isn't a time we just return the results unsorted?

Reply:
@Cadair that would mean that if any single record doesn't have time, or record is empty, then entire result is unsorted, (default as it was earlier). If it is Ok then I will do it using try except on entire yield.

Reply:
> Thanks for the input @jhourcle
> 
> > provideritem might not have a record subblock, or it might have an empty block.
> 
> Cool, I think we handle this.
> 
> Maybe it is worth just guarding this whole block in an `except AttributeError` so that if there isn't a time we just return the results unsorted?

If it were me, I'd assume some default time ... either "now" or the epoch, so they end up either sorting to the end of the list, or the beginning of the list, rather than having the sort fail entirely.  Currently, all items under `record` should have a start time and end time.  (but I'll leave it for the scientists to argue if sorting by start time or the middle of the observation is best)

Reply:
defaulting to "now" (only for sorting) seems reasonable, and start time is good enough rather than calculating the middle :laughing:

Reply:
Seems reasonable. Ok making these changes now.

Reply:
Have made the required checks and changes. Can check.